File: UaRestClient/Model/Measurements/JavaScript/measurements_constants.js
export const NS = 'urn:opcfoundation.org:2024-10:starterkit:measurements';

export const BrowseNames = Object.freeze({
   MeasurementContainerType: 'MeasurementContainerType',
   Orientation: 'Orientation',
   OrientationDataType: 'OrientationDataType',
   Pressure: 'Pressure',
   Reset: 'Reset',
   Temperature: 'Temperature',
});

export const DataTypeIds = Object.freeze({
   OrientationDataType: 'nsu=' + NS + ';i=31',
});

export const MethodIds = Object.freeze({
   MeasurementContainerType_Reset: 'nsu=' + NS + ';i=24',
});

export const ObjectIds = Object.freeze({
   OrientationDataType_Encoding_DefaultBinary: 'nsu=' + NS + ';i=32',
   OrientationDataType_Encoding_DefaultXml: 'nsu=' + NS + ';i=40',
   OrientationDataType_Encoding_DefaultJson: 'nsu=' + NS + ';i=48',
});

export const ObjectTypeIds = Object.freeze({
   MeasurementContainerType: 'nsu=' + NS + ';i=1',
});

export const VariableIds = Object.freeze({
   MeasurementContainerType_Temperature: 'nsu=' + NS + ';i=2',
   MeasurementContainerType_Temperature_EngineeringUnits: 'nsu=' + NS + ';i=7',
   MeasurementContainerType_Temperature_EngineeringUnits_EngineeringUnits: 'nsu=' + NS + ';i=12',
   MeasurementContainerType_Pressure: 'nsu=' + NS + ';i=13',
   MeasurementContainerType_Pressure_EngineeringUnits: 'nsu=' + NS + ';i=18',
   MeasurementContainerType_Pressure_EngineeringUnits_EngineeringUnits: 'nsu=' + NS + ';i=23',
   MeasurementContainerType_Reset_InputArguments: 'nsu=' + NS + ';i=25',
   MeasurementContainerType_Reset_OutputArguments: 'nsu=' + NS + ';i=26',
   MeasurementContainerType_Orientation: 'nsu=' + NS + ';i=27',
});

File: UaWebApiClient/Model/Measurements/Constants/CSharp/measurements_constants.cs
namespace Measurements.WebApi
{
    /// <summary>
    /// The namespaces used in the model.
    /// </summary>
    public static class Namespaces
    {
        /// <remarks />
        public const string Uri = "urn:opcfoundation.org:2024-10:starterkit:measurements";
    }

    /// <summary>
    /// The browse names defined in the model.
    /// </summary>
    public static class BrowseNames
    {
        /// <remarks />
        public const string MeasurementContainerType = "MeasurementContainerType";
        /// <remarks />
        public const string Orientation = "Orientation";
        /// <remarks />
        public const string OrientationDataType = "OrientationDataType";
        /// <remarks />
        public const string Pressure = "Pressure";
        /// <remarks />
        public const string Reset = "Reset";
        /// <remarks />
        public const string Temperature = "Temperature";
    }

    /// <summary>
    /// The well known identifiers for DataType nodes.
    /// </summary>
    public static class DataTypeIds {
        /// <remarks />
        public const string OrientationDataType = "nsu=" + Namespaces.Uri + ";i=31";

        /// <summary>
        /// Converts a value to a name for display.
        /// </summary>
        public static string ToName(string value)
        {
            foreach (var field in typeof(DataTypeIds).GetFields(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static))
            {
                if (field.GetValue(null).Equals(value))
                {
                    return field.Name;
                }
            }

            return value.ToString();
        }
    }

    /// <summary>
    /// The well known identifiers for Method nodes.
    /// </summary>
    public static class MethodIds {
        /// <remarks />
        public const string MeasurementContainerType_Reset = "nsu=" + Namespaces.Uri + ";i=24";

        /// <summary>
        /// Converts a value to a name for display.
        /// </summary>
        public static string ToName(string value)
        {
            foreach (var field in typeof(MethodIds).GetFields(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static))
            {
                if (field.GetValue(null).Equals(value))
                {
                    return field.Name;
                }
            }

            return value.ToString();
        }
    }

    /// <summary>
    /// The well known identifiers for Object nodes.
    /// </summary>
    public static class ObjectIds {
        /// <remarks />
        public const string OrientationDataType_Encoding_DefaultBinary = "nsu=" + Namespaces.Uri + ";i=32";
        /// <remarks />
        public const string OrientationDataType_Encoding_DefaultXml = "nsu=" + Namespaces.Uri + ";i=40";
        /// <remarks />
        public const string OrientationDataType_Encoding_DefaultJson = "nsu=" + Namespaces.Uri + ";i=48";

        /// <summary>
        /// Converts a value to a name for display.
        /// </summary>
        public static string ToName(string value)
        {
            foreach (var field in typeof(ObjectIds).GetFields(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static))
            {
                if (field.GetValue(null).Equals(value))
                {
                    return field.Name;
                }
            }

            return value.ToString();
        }
    }

    /// <summary>
    /// The well known identifiers for ObjectType nodes.
    /// </summary>
    public static class ObjectTypeIds {
        /// <remarks />
        public const string MeasurementContainerType = "nsu=" + Namespaces.Uri + ";i=1";

        /// <summary>
        /// Converts a value to a name for display.
        /// </summary>
        public static string ToName(string value)
        {
            foreach (var field in typeof(ObjectTypeIds).GetFields(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static))
            {
                if (field.GetValue(null).Equals(value))
                {
                    return field.Name;
                }
            }

            return value.ToString();
        }
    }

    /// <summary>
    /// The well known identifiers for Variable nodes.
    /// </summary>
    public static class VariableIds {
        /// <remarks />
        public const string MeasurementContainerType_Temperature = "nsu=" + Namespaces.Uri + ";i=2";
        /// <remarks />
        public const string MeasurementContainerType_Temperature_EngineeringUnits = "nsu=" + Namespaces.Uri + ";i=7";
        /// <remarks />
        public const string MeasurementContainerType_Temperature_EngineeringUnits_EngineeringUnits = "nsu=" + Namespaces.Uri + ";i=12";
        /// <remarks />
        public const string MeasurementContainerType_Pressure = "nsu=" + Namespaces.Uri + ";i=13";
        /// <remarks />
        public const string MeasurementContainerType_Pressure_EngineeringUnits = "nsu=" + Namespaces.Uri + ";i=18";
        /// <remarks />
        public const string MeasurementContainerType_Pressure_EngineeringUnits_EngineeringUnits = "nsu=" + Namespaces.Uri + ";i=23";
        /// <remarks />
        public const string MeasurementContainerType_Reset_InputArguments = "nsu=" + Namespaces.Uri + ";i=25";
        /// <remarks />
        public const string MeasurementContainerType_Reset_OutputArguments = "nsu=" + Namespaces.Uri + ";i=26";
        /// <remarks />
        public const string MeasurementContainerType_Orientation = "nsu=" + Namespaces.Uri + ";i=27";

        /// <summary>
        /// Converts a value to a name for display.
        /// </summary>
        public static string ToName(string value)
        {
            foreach (var field in typeof(VariableIds).GetFields(System.Reflection.BindingFlags.Public | System.Reflection.BindingFlags.Static))
            {
                if (field.GetValue(null).Equals(value))
                {
                    return field.Name;
                }
            }

            return value.ToString();
        }
    }
    
}